apps/studio/src/server/modules/user/__tests__/user.router.test.ts

```diff
@@ -0,0 +1,1001 @@
+import { TRPCError } from "@trpc/server"
+import { resetTables } from "tests/integration/helpers/db"
+import {
+  applyAuthedSession,
+  applySession,
+  createMockRequest,
+} from "tests/integration/helpers/iron-session"
+import {
+  setupAdminPermissions,
+  setupEditorPermissions,
+  setupPublisherPermissions,
+  setupSite,
+  setupUser,
+  setUpWhitelist,
+} from "tests/integration/helpers/seed"
+import { MOCK_STORY_DATE, MOCK_TEST_USER_NAME } from "tests/msw/constants"
+import { beforeEach, describe, expect, it } from "vitest"
+
+import { db, RoleType } from "~/server/modules/database"
+import { createCallerFactory } from "~/server/trpc"
+import { userRouter } from "../user.router"
+
+const createCaller = createCallerFactory(userRouter)
+
+describe("user.router", () => {
+  const TEST_EMAIL = "test@open.gov.sg"
+
+  let caller: ReturnType<typeof createCaller>
+  let session: Awaited<ReturnType<typeof applyAuthedSession>>
+  let siteId: number
+
+  beforeAll(async () => {
+    await setUpWhitelist({ email: TEST_EMAIL })
+  })
+
+  beforeEach(async () => {
+    await resetTables("User", "Site", "ResourcePermission")
+
+    const { site } = await setupSite()
+    siteId = site.id
+    session = await applyAuthedSession()
+    caller = createCaller(createMockRequest(session))
+  })
+
+  describe("create", () => {
+    it("should throw 401 if not logged in", async () => {
+      const unauthedSession = applySession()
+      const unauthedCaller = createCaller(createMockRequest(unauthedSession))
+
+      // Act
+      const result = unauthedCaller.create({
+        siteId,
+        users: [{ email: TEST_EMAIL }],
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+
+    it("should throw 403 if user is not admin of the site", async () => {
```

Comment:
i didn't get this part - we are setting up admin perms for the user in arrange but the assert expects it to throw forbidden for a non-admin user

Reply:
this checks that admin of SiteA cannot create user for SiteB